python/cuml/cuml/cluster/kmeans.pyx

```diff
@@ -784,9 +787,7 @@ class KMeans(Base,
                                        'type': 'dense',
                                        'description': 'Transformed data',
                                        'shape': '(n_samples, n_clusters)'})
-    @deprecate_non_keyword_only("convert_dtype")
-    def fit_transform(self, X, y=None, convert_dtype=False,
-                      sample_weight=None) -> CumlArray:
+    def fit_transform(self, X, y=None, *, convert_dtype=False, sample_weight=None) -> CumlArray:
```

Comment:
I don't think `sample_weight` is meant to be keyword-only here. https://github.com/scikit-learn/scikit-learn/blob/98ed9dc73a86f5f11781a0e21f24c8f47979ec67/sklearn/cluster/_kmeans.py#L1100

```suggestion
    def fit_transform(self, X, y=None, sample_weight=None, *, convert_dtype=False) -> CumlArray:
```